python/plugins/processing/core/GeoAlgorithm.py

```diff
@@ -213,8 +213,16 @@ def execute(self, progress=None, model=None):
             lines = [self.tr('Uncaught error while executing algorithm')]
             lines.append(traceback.format_exc())
             ProcessingLog.addToLog(ProcessingLog.LOG_ERROR, lines)
+            try:
+                message = unicode(e)
+            except Exception:
+                # Encoding the error message failed. Try with the 'replace' mode.
+                try:
+                    message = unicode(e.message, 'utf-8', 'replace')
+                except Exception:
+                    message = self.tr('Unknown error')
```

Comment:
What's the reason for this second catch? In case the error is really unkown (i.e. you have not experienced it yourself) then I'd remove this check, so if it actually gets triggered in the future we can take appropriate steps. In the case the error is not unknown, the error message should be rephrased.